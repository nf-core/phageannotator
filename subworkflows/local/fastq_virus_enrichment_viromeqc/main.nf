//
// Estimate viral enrichment in reads
//

include { VIROMEQC_INSTALL  } from '../../../modules/local/viromeqc/install/main'
include { VIROMEQC_VIROMEQC } from '../../../modules/local/viromeqc/viromeqc/main'

workflow FASTQ_VIRUS_ENRICHMENT_VIROMEQC {
    take:
    fastq_gz                // [ [ meta.id ] , [ 1.fastq.gz, 2.fastq.gz ]  , reads (mandatory)

    main:
    ch_versions = Channel.empty()

    //
    // MODULE: Install ViromeQC index
    //
    ch_viromeqc_index = VIROMEQC_INSTALL ( ).viromeqc_index
    ch_versions = ch_versions.mix(VIROMEQC_INSTALL.out.versions.first())

    //
    // MODULE: Estimate viral enrichment
    //
    ch_enrichment_tsv = VIROMEQC_VIROMEQC ( fastq_gz, ch_viromeqc_index ).enrichment
    ch_versions = ch_versions.mix(VIROMEQC_VIROMEQC.out.versions.first())

    emit:
    enrichment_tsv  = ch_enrichment_tsv // [ [ meta.id ] , enrichment.tsv ]  , viral enrichment estimates
    versions        = ch_versions       // [ versions.yml ]
}
