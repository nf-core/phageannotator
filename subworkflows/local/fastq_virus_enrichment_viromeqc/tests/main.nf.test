nextflow_workflow {

    name "Test subworkflow: FASTQ_VIRUS_ENRICHMENT_VIROMEQC"
    script "subworkflows/local/fastq_virus_enrichment_viromeqc/main.nf"
    workflow "FASTQ_VIRUS_ENRICHMENT_VIROMEQC"

    // Dependencies
    tag "VIROMEQC_INSTALL"
    tag "VIROMEQC_VIROMEQC"


    test("fastq.gz") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
