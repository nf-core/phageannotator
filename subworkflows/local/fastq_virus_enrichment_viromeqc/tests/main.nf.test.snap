{
    "fastq.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c26288b67351c3bdaad635c87a0249e7",
                    "versions.yml:md5,f771871e8cbe81944f6d5fbbc56cc7a5"
                ],
                "enrichment_tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c26288b67351c3bdaad635c87a0249e7",
                    "versions.yml:md5,f771871e8cbe81944f6d5fbbc56cc7a5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-28T17:22:41.423720896"
    }
}
